docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -1,75 +1,151 @@
+Learn how to inspect values during loading, processing and training steps in ML.NET.
 
-This how-to and related sample are currently using **ML.NET version 0.10**. For more information, see the release notes at the [dotnet/machinelearning GitHub repo](https://github.com/dotnet/machinelearning/tree/master/docs/release-notes).
+Data like the one represented below which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView) can be inspected in various ways in ML.NET.
+ 
+```csharp
+HousingData[] housingData = new HousingData[]
+{
+    new HousingData
+    {
+        Size = 600f,
+        HistoricalPrices = new float[] { 100000f ,125000f ,122000f },
+        CurrentPrice = 170000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 200000f, 250000f, 230000f },
+        CurrentPrice = 225000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 126000f, 130000f, 200000f },
+        CurrentPrice = 195000f
+    },
+    new HousingData
+    {
+        Size = 850f,
+        HistoricalPrices = new float[] { 150000f,175000f,210000f },
+        CurrentPrice = 205000f
+    },
+    new HousingData
+    {
+        Size = 900f,
+        HistoricalPrices = new float[] { 155000f, 190000f, 220000f },
+        CurrentPrice = 210000f
+    },
+    new HousingData
+    {
+        Size = 550f,
+        HistoricalPrices = new float[] { 99000f, 98000f, 130000f },
+        CurrentPrice = 180000f
+    }
+};
+```
 
-During the experiment, you may want to observe and validate the data processing results at a given point. This isn't easy since ML.NET operations are lazy, constructing objects that are 'promises' of data.
+## Convert IDataView to IEnumerable
 
-The `GetColumn<T>` extension method lets you inspect the intermediate data. It returns the contents of one data column as an `IEnumerable`.
+One of the simplest ways to inspect the values of an IDataView is to convert it to an [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1). To convert an [`IDataView`](xref:Microsoft.ML.IDataView) to [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1) use the [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) method.
 
-The following example shows how to use the `GetColumn<T>` extension method:
+> [!WARNING]
+> [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) will load the entire [`IDataView`](xref:Microsoft.ML.IDataView) into memory which may affect performance.
 
-[Example file](https://github.com/dotnet/machinelearning/tree/master/test/data/adult.tiny.with-schema.txt):
+```csharp
+// Create an IEnumerable from IDataView
+HousingData[] housingDataEnumerable =
+    mlContext.Data.CreateEnumerable<HousingData>(data, reuseRowObject: false).ToArray();
```

Comment:
Can we add a note on when you would set reuseRowObject to true?

Reply:
#resolved

Reply:
reuseRowObject = true, is good for efficiency when you don't materialize the IEnumerable. You need to set to false here, because .ToArray() at the end is materializing the IEnumerable. 

If you're sure the the IEnumerable will never be materlialized, reuseRowObject can be set to true for saving memory allocation.

---
In reply to: [278766994](https://github.com/dotnet/docs/pull/11813#discussion_r278766994) [](ancestors = 278766994)